tools/rosbag/src/player.cpp

```diff
@@ -227,15 +236,20 @@ void Player::publish() {
             time_publisher_.setPublishFrequency(-1.0);
 
         paused_time_ = now_wt;
-
-        // Call do-publish for each message
+        
         foreach(MessageInstance m, view) {
             if (!node_handle_.ok())
                 break;
 
             doPublish(m);
+
+            if (skip_time_)
+                break;
```

Comment:
Why wouldn't this either wrap around the `foreach` as an `if` statement or at least `break` before the first `doPublish`? Put another way, why does this only do one `doPublish` when `skip_time_` is `true`?